Lesson-2/BankAcc.cs

```diff
@@ -113,6 +113,36 @@ public override string ToString()
         {
             return $"Тип счёта: {this.Type}\nНомер счёта: {this.Numb}\nБаланс: {this.Balance}";
         }
+
+        public static bool operator ==(BankAcc first, BankAcc second)
+        {
+            return first.Balance== second.Balance;
+        }
+
+        public static bool operator !=(BankAcc first, BankAcc second)
+        {
+            return !(first.Balance == second.Balance);
+        }
+
+        public bool Equals(BankAcc compared)
+        {
+            return this.Balance == compared.Balance;
+        }
+
+        public override bool Equals(object? obj)//есть здесь, только потому что в задании было сказано "переопределить", не "перезагрузить". 
+        {
+            if(obj == null)
+                return false;
+            var notObj = obj as BankAcc;
+            if (notObj is null)
+                return false;
+            return this.Balance == notObj.Balance;
+        }
+
+        public override int GetHashCode()
+        {
+            return int.Parse(this.Numb).GetHashCode();
```

Comment:
Это ужасно...

Reply:
```suggestion
            return Numb.GetHashCode();
```

Reply:
Хотите что бы у счёта хеш-код всегда был одинаковы если используется один и тот же номер счёта, то
```suggestion
            var hash = 312;
            foreach(var c in Numb)
                unchecked
                {
                    hash = (hash * 312) ^ c.GetHashCode()
                }
            return hash;
```
Это будет работать медленнее...